universal-application-tool-0.0.1/app/services/question/QuestionService.java

```diff
@@ -0,0 +1,60 @@
+package services.question;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Locale;
+import java.util.Optional;
+
+public interface QuestionService {
+
+  /**
+   * Creates a new Question Definition. Returns a QuestionDefinition object on success and null on
+   * failure.
+   *
+   * <p>This will fail if: - The path provided already resolves to a QuestionDefinition or Scalar.
+   *
+   * <p>NOTE: This does not update the version.
+   */
+  public abstract Optional<QuestionDefinition> create(QuestionDefinition definition);
+
+  /**
+   * Adds a new translation to an existing question definition. Returns true if the write is
+   * successful.
+   *
+   * <p>The write will fail if: - The path does not resolve to a QuestionDefinition. - A translation
+   * with that Locale already exists for a given question path.
+   *
+   * <p>NOTE: This does not update the version.
+   */
+  public abstract boolean addTranslation(
+      String path, Locale Locale, String questionText, Optional<String> questionHelpText);
```

Comment:
Question: why `path` rather than the qd id?

Reply:
I guess I'm viewing it from an API perspective. Path will always be more human readable, and if I'm hand writing these (without a user interface) I'd much rather write `path: 'applicant.name'` than `id: 'fhwdgads'`

It does raise a question though... once we implement versioning we'll have

| id                | path  | version |
| ----------- | ----------- | ----------- 
| fhwgads      | applicant.name | 1
| hgnsdhjsd   | applicant.name | 2

If an admin wants to add another translation to `applicant.name`will it only be available in version 2 or should it be available to both?

Remember, this is strictly for `addTranslation`. If we already have an existing translation, we would need to use `update` and increment the version of the question.

Reply:
I'm hesitant to think too far into versioning yet. That being said, my guess is that when we create a new version we'll copy all of the questions from the previous version so they can be edited/deleted totally independent of the previous one. I don't know what the right behavior is with locale updates specifically 🤔 . I'm inclined to hold off on making decisions around it since it's not in the current milestone if we can.

Reply:
Additional reasoning, I also wanted to enforce the notion that an admin can add translations without having to re-version of the question. (I believe that this is the only question update that we can safely do without re-versioning.)

Reply:
That's a good call.

Reply:
I'm fine with leaving it path for now but think whoever implements locale updates should have the right to change it to whatever is most convenient